tuning/autotune.py

```diff
@@ -519,56 +530,77 @@ def benchmark_compiled_candidates(
     """Benchmark the candidate files and store the top20 results in file (best.log). Return the log file"""
     logging.info("benchmark_top_candidates()")
 
-    task_list = []
-    for compiled_file in compiled_files:
-        command = ["./benchmark_dispatch.sh", f"{compiled_file}"]
-        check = False
-        task_list.append((args, command, check))
-
-    worker_context_queue = create_worker_context_queue(args.devices)
-    results = multiprocess_progress_wrapper(
-        num_worker=len(args.devices),
-        task_list=task_list,
-        function=worker_run_command_with_device_id,
-        initializer=init_worker_context,
-        initializer_inputs=(worker_context_queue,),
-    )
-
-    benchmark_results = [result.stdout for result in results]
-
     results_log = base_dir / "results.log"
-    with results_log.open("w") as log_file:
-        log_file.writelines(benchmark_results)
-
+    best_log = base_dir / "best.log"
+    benchmark_results = []
     best_results = []
-    with results_log.open("r") as log_file:
-        for line in log_file:
-            if "failed" not in line:
-                parts = line.split()
 
-                # Update candidate tracker
-                candidate_trackers[int(parts[0])].first_benchmark_time = float(
-                    parts[-1]
-                )
+    if args.dry_run:
+        # generate random top candidates list from compiled_files
+        benchmark_list_len = (
+            random.randint(20, len(compiled_files))
+            if len(compiled_files) > 20
+            else len(compiled_files)
+        )
+        random_candidate_indices = set()
+        while len(random_candidate_indices) < benchmark_list_len:
+            random_candidate_indices.add(random.randint(0, len(compiled_files) - 1))
+        random_candidate_indices = list(random_candidate_indices)
+
+        # genereate random benchmark time
+        benchmark_results = ["" for _ in range(len(compiled_files))]
+        with tqdm(total=benchmark_list_len) as pbar:
+            for i in random_candidate_indices:
+                candidate_id = re.search(
+                    r"/(\d+)\.vmfb$", str(compiled_files[i])
+                ).group(
+                    1
+                )  # use compiled_file path string to extract candidate_id
+                benchmark_results[
+                    i
+                ] = f"{candidate_id}\tMean Time: {random.randint(50, 500):.1f}\n"
+                pbar.update(1)
```

Comment:
We should be able to write a helper function that generates fake data and test it. Also, no need to use a progress bar here.